variables.tf

```diff
@@ -3,11 +3,22 @@ variable "aws_sqs_queue_name" {
   description = "Name of the AWS SQS queue to create and to listen to."
 }
 
-variable "aws_lambda_function_name" {
```

Comment:
This is a breaking change and will require a lot of changes done in microservices that already use this Terraform module.

- So far, using various Terraform modules from different authors it seems that people most often provide plain values rather than whole objects as module arguments.
- The user of the module might not necessarily have the Lambda function object at their disposal. Perhaps the Lambda is defined in another project and they only have its name and role name? This is not a scenario I would endorse but nevertheless it can exist.
- Someone may need to use this module for a Lambda function that's not even defined in Terraform.

That is why function name and function role provided as separate arguments are more verbose.

The 404 issue can be easily resolved by using a `depends_on` clause when using this module in your client code.